recipes/ms-gsl/all/conanfile.py

```diff
@@ -50,8 +57,14 @@ def package_id(self):
         self.info.header_only()
 
     def package_info(self):
+        if int(self.version.split(".")[0]) > 2:
+            if self.options.on_contract_violation != 'terminate':
+                raise ConanInvalidConfiguration("Microsoft GSL only supports terminate on contract violation since version 3.0.0")
+
         self.cpp_info.defines = [
             self._contract_map[str(self.options.on_contract_violation)]
         ]
-        self.cpp_info.names["cmake_find_package"] = "Microsoft.GSL"
-        self.cpp_info.names["cmake_find_package_multi"] = "Microsoft.GSL"
+          
+        if int(self.version.split(".")[0]) > 2:
+            self.cpp_info.names["cmake_find_package"] = "Microsoft.GSL"
```

Comment:
Why it's only available for 3.0.0 now?

Reply:
https://github.com/conan-io/conan-center-index/pull/1388#discussion_r410915349